### PR TITLE
Updates to release checklist and PR template to improve processes

### PR DESCRIPTION
**Pull request by Aljullu**
This PR introduces several changes to the release checklist:

* States that release leads should ask porters from both teams to test the release.
* Gives some steps to proceed when PRs are not working as expected.
* _Publish Posts_ section has been moved to the bottom of the process. So it's more clear that this step doesn't block creating the PR in WC core.
* Mentions that porters are responsible of providing the release lead with the highlights of the changes their team has introduced in that release.

In general, this PR reduces the amount of work the release lead needs to do with the con of introducing some more work for porters. The rationale is that porters from each team are more aware of what their team is working on, so for them it's easier to test and provide release post insights than it is for the release leads.

### CODE REVIEW

File: .github/patch-initial-checklist.md
Comment (ralucaStan) [inline .github/patch-initial-checklist.md]: I think we need to mention here that we need to revert the PR's commits

File: .github/release-initial-checklist.md
Comment (ralucaStan) [inline .github/release-initial-checklist.md]: Same comment, I think we need to mention here that we need to revert the PR's commits

File: .github/patch-initial-checklist.md
Comment (ralucaStan) [inline .github/patch-initial-checklist.md:90]: Should we do a final proof read with both teams before posting? From what I've noticed there are usually small issues that are fixed in this step, like versions, links etc

File: .github/patch-initial-checklist.md
Comment (Aljullu) [reply .github/patch-initial-checklist.md]: Ah, right. Better to mention it. Done in 09c929eabe74ebe8dd94eca29f467da2b7b5fa6f.

File: .github/release-initial-checklist.md
Comment (Aljullu) [reply .github/release-initial-checklist.md]: Done in 09c929eabe74ebe8dd94eca29f467da2b7b5fa6f.

File: .github/patch-initial-checklist.md
Comment (Aljullu) [reply .github/patch-initial-checklist.md:90]: I was thinking having the porters write the highlights and the release lead merging them in the post would be enough. But I don't have a strong opinion.

**Comment by ralucaStan** [comment]
@Aljullu I've added my changes

Release template
- the Write testing notes has been moved to the PR template, and removed from the release checklist
- mention that common functionality should be tested by both porters
- update the instructions on how to create the WC PR's changelog

Pull request template
- better visibility for the feature flag/experimental changes
- add before/after table for screenshots
- add WooCommerce Visibility to help with the changelog of the WC PR
- add an extra check to not include a PR in The testing notes

File: .github/patch-initial-checklist.md
Comment (ralucaStan) [inline .github/patch-initial-checklist.md]: There is a clash between this section and the one above from Create the Testing Notes that states

```
* [ ] Smoke test built release zip using the testing instructions you created:
  * At least one other person should test the built zip - ping the current Rubik porter to be this person.
```

**Comment by ralucaStan** [comment]
I haven't removed this part, I feel we should get the teams aligned before doing this:

```
* [ ] Remind whoever is porter this week to audit our codebase to ensure this [experimental interface document](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/docs/blocks/feature-flags-and-experimental-interfaces.md) is up to date. See Pca54o-rM-p2 for more details.
```

File: .github/patch-initial-checklist.md
Comment (ralucaStan) [reply .github/patch-initial-checklist.md]: I think the changes we made `Ask the porter of Rubik and Kirigami ...` should replace this entry: 
```* At least one other person should test the built zip - ping the current Rubik porter to be this person.```

@Aljullu what do you think?

**Comment by tomasztunik** [comment]
Should we put any timeboxing or priority highlight against other work porters have on their agenda? I remember we had this revelation moment when Michael shared our success rate at delivering on schedule. Maybe this should be included in some form to support being on time?

Though I guess it belongs more to Porter's Handbook :)

File: .github/patch-initial-checklist.md
Comment (Aljullu) [reply .github/patch-initial-checklist.md]: Ah, good catch. Yes, I think it makes sense. Are you going to update it or do you want me to do it?

File: .github/pull_request_template.md
Comment (Aljullu) [inline .github/pull_request_template.md]: Do you know what's the difference between `Manual Testing` and `User Facing Testing`? I wonder if we can take the chance and unify those two sections.

**Review by Aljullu — COMMENTED** [review]
Changes look good, thanks @ralucaStan!

> I haven't removed this part, I feel we should get the teams aligned before doing this:

Agree, this can be discussed somewhere else.

> Should we put any timeboxing or priority highlight against other work porters have on their agenda?

Right, I think releases should be one of the top priorities for porters. But as you mentioned, I think that belongs to the Team Handbook (we can update them if this PR ends up getting merged).

File: .github/patch-initial-checklist.md
Comment (ralucaStan) [reply .github/patch-initial-checklist.md]: @Aljullu it would be great if you could handle this change in both initial and patch templates

File: .github/pull_request_template.md
Comment (ralucaStan) [reply .github/pull_request_template.md]: This is a great point. 
Manual testing was for the instructions from us, maintainers, this is where branch/code changes would go in order to test. etc
User Facing Testing would be the nicely, end-user facing testing steps would go. 

I guess we can unify and have only something along the lines of 
```
### Testing steps

<!-- Check if the change is not relevant to the "user" (merchant) or the testing requires branch / code changes -->
* [ ] Don't include in the release Testing Notes

<!-- If the above check is not made, then write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->

1.
2.
3.
```

Comment (Aljullu) [reply .github/pull_request_template.md]: > I guess we can unify and have only something along the lines of

:100:

File: .github/patch-initial-checklist.md
Comment (Aljullu) [reply .github/patch-initial-checklist.md]: Updated in 2c75cf86adc7158d8c1841ab755274eeeaf94b66. Let me know how it looks.

File: .github/patch-initial-checklist.md
Comment (nielslange) [inline .github/patch-initial-checklist.md]: I wonder in how far the team porter is a good fix for fixing a blocker. I was thinking that fixing a blocker would be the responsibility of the PR author only.

**Review by nielslange — APPROVED** [review]
Pre-approving this PR. I only added a comment regarding fixing a blocker. Personally, I see the PR author as the only responsible person for fixing a blocker.

File: .github/patch-initial-checklist.md
Comment (Aljullu) [reply .github/patch-initial-checklist.md]: Good point. I rewrote it in 329906d53cf925d91e6d8b9ba093e0d3c2b06635, let me know how it looks now! :slightly_smiling_face:

Comment (nielslange) [reply .github/patch-initial-checklist.md]: Looks great now, @Aljullu!

File: .github/patch-initial-checklist.md
Comment (ralucaStan) [inline .github/patch-initial-checklist.md]:
```suggestion
* [ ] Ask the porters of Rubik and Kirigami to smoke test the built release zip as well and to approve the PR if everything looks good. 

Each porter is responsible for testing the PRs that fall under the focus of their own team. Shared functionality should be tested by both porters. This means that the Rubik porter will mostly test Cart & Checkout changes while Kirigami porter will test the other blocks.
```

File: .github/release-initial-checklist.md
Comment (ralucaStan) [inline .github/release-initial-checklist.md]:
```suggestion
* [ ] Ask the porters of Rubik and Kirigami to test the built zip as well and to approve the PR if everything looks good.

Each porter is responsible for testing the PRs that fall under the focus of their own team. Shared functionality should be tested by both porters. This means that the Rubik porter will mostly test Cart & Checkout changes while Kirigami porter will test the other blocks.
```

**Review by ralucaStan — APPROVED** [review]
@Aljullu the changes look good, I am approving this PR, hope this changes will be helpful.  
I left the small comments to increase the clarity and readability for the porter testing section

**Comment by ralucaStan** [comment]
@Aljullu I added another line to highlight that it is not the release lead responsibility to log any minor issues that appear during testing. fa07f7cfb77c060e5eeee290127f598331ac2579 

They fall under this step `Take care of any communication around the release (as noted in the release steps documentation for the product).` from PdToLP-O-p2. If you think we should go more in detail there as well let's have a chat.

File: .github/patch-initial-checklist.md
Comment (nielslange) [inline .github/patch-initial-checklist.md]: Nit-pic: Given that all other sentences end with a fullstop, this one should also end with one.

File: .github/release-initial-checklist.md
Comment (nielslange) [inline .github/release-initial-checklist.md]: See previous comment.

**Comment by Aljullu** [comment]
> @Aljullu I added another line to highlight that it is not the release lead responsibility to log any minor issues that appear during testing. https://github.com/woocommerce/woocommerce-gutenberg-products-block/commit/fa07f7cfb77c060e5eeee290127f598331ac2579

Makes sense to me, thanks! :+1:

File: .github/pull_request_template.md
Comment (Aljullu) [reply .github/pull_request_template.md]: @ralucaStan are you ok if I implement this change and merge `Manual Testing` and `User Facing Testing` into a `Testing steps` section?

Comment (ralucaStan) [reply .github/pull_request_template.md]: Sure, thanks for taking over here.

Comment (Aljullu) [reply .github/pull_request_template.md]: Done in 45cc99075e62813bdefdee856922330f75a6ef36.

Comment (ralucaStan) [reply .github/pull_request_template.md]: Looks great 👍

**Comment by Aljullu** [comment]
@ralucaStan can I get another review from you? I made a couple of changes:

* Specified the milestone should be closed as the first step.
* Removed this step: `Remove any unfinished issues from the Zenhub epics completed by this release and then close the epics.`. I don't think the release lead should be in charge of closing epics, that's a task for the DRI of each project.

File: .github/patch-initial-checklist.md
Comment (ralucaStan) [inline .github/patch-initial-checklist.md:67]: Should we include here the step of creating a new milestone for the next version?

**Comment by ralucaStan** [comment]
> @ralucaStan can I get another review from you? I made a couple of changes:
> 
> * Specified the milestone should be closed as the first step.


@Aljullu I left a [comment](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/6293#discussion_r860855082) for the milestone changes, let me know what you think

> * Removed this step: `Remove any unfinished issues from the Zenhub epics completed by this release and then close the epics.`. I don't think the release lead should be in charge of closing epics, that's a task for the DRI of each project.

Nicely spotted, totally agree. This will also prevent some projects for being closed without an updated description

File: .github/patch-initial-checklist.md
Comment (Aljullu) [reply .github/patch-initial-checklist.md:67]: Good point. I'm a bit confused by why that step wasn't there before. :confused: 

I added it in 7776d882bc858747a9eee18d674f011cb1959bff. I added it to the top of the list, so the next milestone is created right after closing the current one. Otherwise, there might be some time when there is no milestone open.

**Comment by Aljullu** [comment]
I will go ahead and merge this PR. If we want to make any more changes, they can be done in a follow-up PR.

**Pull request merged**
